### DISCUSSION TITLE
Downloading 'latest' photos does not seem to work for me...

### DISCUSSION THREAD

**Discussion by vadalus911** [icloud-photos-downloader/icloud_photos_downloader#632] [Q&A]
Hi Guys,

I am running this in docker. I have a phone with 23k photos, ive done a one off download and now i am just looking to download updates. I was hoping to avoid rescanning the whole library everytime and it looks as though the --until-found=x option should do exactly what i want. Unfortunately it does not seem to be checking the LATEST x number of photos. From the logs it looks as though it just checks ANY x photos which have been downloaded and then stops. So for me this does not work, it appears to check old photos, realizes they've already been downloaded and thats it.

Is this is known bug or this is supposed to be how it works?

Also, I was hoping to also keep down the amount of logging, at the moment it's logging 23k lines of 'photo already downloaded'. Ideally it would not log that but only ACTUAL photo downloads. Again, is this possible?

**Comment by AndreyNikiforov** [comment]
> ... Unfortunately it does not seem to be checking the LATEST x number of photos. From the logs it looks as though it just checks ANY x photos which have been downloaded and then stops. So for me this does not work, it appears to check old photos, realizes they've already been downloaded and thats it.
> 
> Is this is known bug or this is supposed to be how it works?

`until-found` flag forces `icloudpd` to look to photos in iCloud ordered by `date_added` field descending, not by `date_created` which is used for folder structure. Once specified number of photos are found locally, `icloudpd` assumes that the rest were downloaded previously. This incremental mechanism will never fill the "gaps" in your history of course.

**Reply by vadalus911** [reply]
Hi thanks

must be some idiosyncrasy with the way my photos have been synced between phone upgrades maybe? In my test It logged X previously downloaded files from 2022 before Quiting out. I’m not sure how these could possibly have any date field which is more recent that a photo I’ve just taken on the phone and synced to iCloud …

**Reply by AndreyNikiforov** [reply]
If you want to keep a copy of your iCloud on your local drive (==your goal), then your focus will most likely be a) a reliability of `icloudpd` in keeping `iCloud` synchronized and b) delay/lag between files appearing in `icloud` and on local device. With these two in mind, running `icloudpd` with `--watch-with-interval` and without `-until-found` may make more sense.

**Reply by vadalus911** [reply]
Many thanks. That is what I’m doing. I was just looking for a few optimisations, none of which are show stoppers.
